registry/remote/example_test.go

```diff
@@ -181,6 +185,56 @@ func ExampleRepository_Push() {
 	// Push finished
 }
 
+// ExampleRepository_Push_referenceManifest gives an example snippet for pushing a reference manifest.
+func ExampleRepository_Push_referenceManifest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
+
+	// 1. assemble the referenced manifest
+	rfedManifest := artifactspec.Manifest{
+		MediaType:    artifactspec.MediaTypeArtifactManifest,
+		ArtifactType: "referenced manifest",
+	}
+	rfedManifestContent, _ := json.Marshal(rfedManifest)
+	rfedManifestDescriptor := artifactspec.Descriptor{
+		MediaType:    ocispec.MediaTypeImageManifest,
+		ArtifactType: "referenced manifest descriptor",
+		Digest:       ocidigest.FromBytes(rfedManifestContent),
+		Size:         int64(len(rfedManifestContent)),
+	}
+
+	// 2. assemble the referencing artifact manifest
+	rfingManifest := artifactspec.Manifest{
+		MediaType:    artifactspec.MediaTypeArtifactManifest,
+		ArtifactType: "referencing manifest",
+		Subject:      rfedManifestDescriptor,
+	}
+	rfingManifestContent, _ := json.Marshal(rfingManifest)
+	rfingManifestDescriptor := artifactspec.Descriptor{
+		MediaType:    ocispec.MediaTypeImageManifest,
+		ArtifactType: "referencing manifest descriptor",
+		Digest:       ocidigest.FromBytes(rfingManifestContent),
+		Size:         int64(len(rfingManifestContent)),
+	}
+
+	// 3. push the manifest descriptor and content
+	err = repo.Push(ctx, descriptor.ArtifactToOCI(rfingManifestDescriptor), bytes.NewReader(rfingManifestContent))
```

Comment:
This won't work. Will failed the registry-side checking with below error since `rfedManifest` is not in the remote registry yet. **And yes that's why we need oras.Copy!**

400: PUT "https://$registry/v2/example/manifests/sha256:aa477aa71f0a7f5a8339ac982d30569a4200dbef53aeae77e1df6c9999eb92be": unexpected status code 400: manifest invalid: manifest invalid"

So this should be an example for copy.

P.S. We can reuse this example for extended copy via copying the referee.

Reply:
There will be a separate example for copying artifact reference manifest (See #60). For this example, we need to push the subject manifest and then the referrer artifact.